src/main/webapp/resources/js/pages/projects/samples-metadata-import2/components/SampleMetadataImportReview.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from "react";
+import { navigate } from "@reach/router"
+import {
+  Button,
+  Space,
+  Table,
+  Tag,
+  Typography,
+} from "antd";
+import { SampleMetadataImportWizard } from "./SampleMetadataImportWizard";
+import { useGetProjectSampleMetadataQuery } from "../../../../apis/metadata/metadata-import";
+import { IconArrowLeft, IconArrowRight } from "../../../../components/icons/Icons";
+
+const { Text } = Typography
+
+/**
+ * React component that displays Step #3 of the Sample Metadata Uploader.
+ * This page is where the user reviews the metadata to be uploaded.
+ * @returns {*}
+ * @constructor
+ */
+export function SampleMetadataImportReview({ projectId }) {
+  const [selected, setSelected] = useState([]);
+  const { data } = useGetProjectSampleMetadataQuery(projectId);
```

Comment:
In the api itself, you can give the query a `transformResponse` method, within that you can modify the response.   I have set up a utility method for adding keys to lists (btw just and index should never be used as a key because there can be other things with the same index), it is called `addKeysToList`.  You can see how to set this up in `apis/metadata/fields.js` on line 32.  This will straight out make your data, your dataSource 😀, and you can remove the declaration on line 26.